### PR TITLE
#368 Test org.jpeek.web.ReportsTest#rendersOneReport is unstable

### PR DESCRIPTION
**Pull request by dgarus**
https://github.com/yegor256/jpeek/issues/368

Added extracting classes from Jar file using java API

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#385](https://github.com/yegor256/jpeek/issues/385) is assigned to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/921f55a9-6d97-47b5-bec7-af449d98e6dc); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by 0crat** [comment]
@fanifieiev[/z](https://www.0crat.com/u/fanifieiev) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

File: src/main/java/org/jpeek/web/Reports.java [hunk 1]
```diff
@@ -151,6 +130,34 @@
...
+        final JarFile jar = new JarFile(origin);
+        final Enumeration<JarEntry> entries = jar.entries();
+        while (entries.hasMoreElements()) {
+            final JarEntry entry = entries.nextElement();
+            final File item = new File(dest + File.separator + entry.getName());
+            if (entry.isDirectory()) {
+                item.mkdir();
+                continue;
+            }
+            try (
```
Comment (fanifieiev) [inline src/main/java/org/jpeek/web/Reports.java]: @dgarus I think you can leverage `Cactoos `io package to reduce the below code.
Look at the Cactoos `TeeInput`

**Comment by dgarus** [comment]
@fanifieiev Hi!
I fixed your comments.
Thank you!

File: src/main/java/org/jpeek/web/Reports.java [hunk 2]
```diff
@@ -151,6 +128,34 @@
...
+        try (JarFile jar = new JarFile(path.toFile())) {
+            final Enumeration<JarEntry> entries = jar.entries();
+            while (entries.hasMoreElements()) {
+                final JarEntry entry = entries.nextElement();
+                final Path item = path.getParent().resolve(entry.getName());
+                if (entry.isDirectory()) {
+                    item.toFile().mkdir();
+                    continue;
+                }
+                if (!item.getParent().toFile().exists()) {
```
Comment (fanifieiev) [inline src/main/java/org/jpeek/web/Reports.java]: @dgarus Please extract 'item.getParent().toFile()' to a variable, so that you do not need to repeat it.

**Comment by dgarus** [comment]
@fanifieiev Fixed.
Thank you!

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job audit: performer @fanifieiev[/z](https://www.0crat.com/u/fanifieiev)[/z](https://www.0crat.com/u/fanifieiev) didn't make CR comments

**Comment by 0crat** [comment]
Code review was too long (12 days), architects (@paulodamaso) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Pull request merged**